File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/domain/shared/valueobjects/Uf.java
package br.com.deltaglobalbank.customers.domain.shared.valueobjects;

import java.util.Locale;

public record Uf(String value) {
    public Uf {
        String uf = value.trim().toUpperCase(Locale.ROOT);
        if (!uf.matches("^[A-Z]{2}$")) {
            throw new IllegalArgumentException("uf_invalid");
        }
        value = uf;
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/domain/shared/valueobjects/Uf.kt
package br.com.deltaglobalbank.customers.domain.shared.valueobjects

@JvmInline
value class Uf private constructor(val value: String) {
    companion object {
        operator fun invoke(raw: String): Uf {
            val uf = raw.trim().uppercase()
            require(REGEX.matches(uf)) { "uf_invalid" }
            return Uf(uf)
        }
        private val REGEX = Regex("^[A-Z]{2}$")
    }
}